doozer/doozerlib/cli/scan_fips.py

```diff
@@ -0,0 +1,55 @@
+"""
+For this command to work, https://github.com/openshift/check-payload binary has to exist in PATH and run as root
+"""
+import json
+
+import click
+import koji
+from doozerlib.cli import cli, pass_runtime
+from doozerlib.runtime import Runtime
+from typing import Optional
+from doozerlib.exectools import cmd_gather
+
+
+class ScanFipsCli:
+    def __init__(self, runtime: Runtime, nvrs: Optional[list]):
+        self.runtime = runtime
+        self.nvrs = nvrs
+
+        # Initialize runtime and brewhub session
+        self.runtime.initialize(clone_distgits=False)
+        self.koji_session = koji.ClientSession(self.runtime.group_config.urls.brewhub)
+
+    def run(self):
+        # Get the list of NVRs to scan for
+        # (nvr, pull-spec) list of tuples
+        image_pullspec_mapping = []
+
+        for nvr in self.nvrs:
+            # Find the registry pull spec
+            build_info = self.koji_session.getBuild(nvr)
+
+            # Eg registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-sriov-network-operator@sha256:da95750d31cb1b9539f664d2d6255727fa8d648e93150ae92ed84a9e993753be
+            # from https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=2777601
+            pull_spec = build_info["extra"]["image"]["index"]["pull"][0]
+            image_pullspec_mapping.append((nvr, pull_spec))
+
+        problem_images = {}
+        for nvr, pull_spec in image_pullspec_mapping:
```

Comment:
It might be worth (as this is an easy change) to use `cmd_gather_async` instead, and run all these commands in a pool of async tasks. Something like:

```

class ScanFipsCli:

   ...

    async def check_payload(self, pullspec, problems):
        rc, _, _ = await cmd_gather_async(f"check-payload scan image --spec {pull_spec}")
        if rc != 0:
            problems[nvr] = pull_spec

    async def run(self):
        ...

        problem_images = {}
        await asyncio.gather(*[self.check_payload(pullspec, problem_images) 
                               for nvr, pull_spec in image_pullspec_mapping])
```

Reply:
Thank you! Added in https://github.com/openshift-eng/art-tools/pull/171/commits/29c790f396924732039cd3e027ca8a355e59d936